Refactor card-quote

lcummings12 opened the pull request:
## Summary

Refactored the `card-quote.html` short code to include the blockquote component classes and align its structure more closely with the `note.html` style.

## Preview

[Link to Preview]()

<!--
⚠️ Significant visual changes require submitting previous design to wayback machine.
-->

## Solution

The solution involved refactoring the `card-quote.html` short code to improve its structure and ensure the `blockquote` element has appropriate component classes.

- Extracted `$bg`, `$cite`, and `$text` as variables at the beginning.
- Added `card-quote__blockquote` class to the `blockquote` element.
- Ensured the logic and structure align more closely with the `note.html` style.




## How To Test
The changes made were solely refactoring. If properly implemented, there should no visual change. 
| Proposed change                                                                                  | Production                                                                                       |
| ------------------------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------ |
| [Proposed change](http://localhost:1313/2024/07/08/a-small-teams-journey-through-digital-maturity/) | [Production](https://digital.gov//2024/07/08/a-small-teams-journey-through-digital-maturity/) |




<!--
## Dependency updates

| Dependency name              | Previous version | New version |
| ---------------------------- | :--------------: | :---------: |
| [Updated dependency example] |     [1.0.0]      |   [1.0.1]   |
| [New dependency example]     |        --        |   [3.0.1]   |
| [Removed dependency example] |     [2.10.2]     |     --      |
-->

<!--
Before opening this PR, make sure you’ve done whichever of these applies to you:
- [ ] Branch is up-to-date and includes latest from `main`
- [ ] PR has correct labels
- [ ] A11y testing (voice over testing, meets WCAG, run axe tools)
-->

[reg-testing.md](https://github.com/user-attachments/files/16533374/reg-testing.md)

nick-mon1 commented:
@lcummings12 This improvement is great, don't forget to update the `_card-quote.scss` with the new class names.

lcummings12 commented:
@nick-mon1  Ended up removing that class as it served no purpose other than being consistent with the model file.

nick-mon1 commented on themes/digital.gov/layouts/shortcodes/card-quote.html:
```diff
@@ -1,11 +1,26 @@
-{{- if (.Get "text") -}}
-  <div class="card-quote {{ if (.Get "bg") -}}{{- .Get "bg" -}}{{- end -}}">
+{{/* card-quote short code to display a block of quoted text with optional citation
+
+  @param {string} bg - optional value of "dark" to display a dark background color otherwise will display as gold
+  @param {string} cite - optional string to display a citation
+  @param {string} text - required text for displaying a quote
+
+  @example usage
+  {{< card-quote text="The big idea goes here" cite="Dr. Meredith Larson, Department of Education"
+>}}
+
+*/}}
```
### Improvement

Added comments to explain the component.

nick-mon1 commented on themes/digital.gov/layouts/shortcodes/card-quote.html:
### Suggestion

This component lives in the short code directory, we could move this to partials and refactor this file to use the partial.

Why? This will keep our files more organized and consistent.

nick-mon1 commented on themes/digital.gov/layouts/shortcodes/card-quote.html:
```diff
@@ -1,11 +1,26 @@
-      >
-      {{- if (.Get "cite") -}}
-        <cite>— {{- .Get "cite" | markdownify -}}</cite>
+      <span class="card_quote__quotation-mark">“</span>
```
### Improvement

I've renamed `q` to prefer BEM naming convention and for clarity.

nick-mon1 commented on themes/digital.gov/layouts/shortcodes/card-quote.html:
```diff
@@ -1,11 +1,26 @@
+{{ $text := (.Get "text") }}
+
+{{- if $text -}}
+  <div class="card-quote {{ if $bg }}card-quote--{{ $bg }}{{ end }}">
```
### Improvement

This uses the BEM naming convention for variants.

lcummings12 commented:
@nick-mon1 Improved the implementation as you suggested by restructuring into a shortcode / partial setup where the `quote-block.html` partial calls the `quote-block.html` shortcode. Additionally, renamed and reformatted `card-quote.js` to support the new setup with `quote-block`.  Switched out all instances of the original shortcode in the .md files with the updated version.  For testing purpose, I created regression file with the variants I could think of. I attached it to the PR, but here is an image of it.
<img width="552" alt="image" src="https://github.com/user-attachments/assets/0043e2c4-5fb4-4abf-a12f-a8eff8dc29c1">

RileySeaburg reviewed (dismissed):
Great attention to detail on this.

nick-mon1 reviewed (approved):
Thanks @lcummings12 for these updates. I fixed a couple of the styles to match font styles, all set.
